src/main/java/teammates/common/datatransfer/GeneralLogEntry.java

```diff
@@ -0,0 +1,114 @@
+package teammates.common.datatransfer;
+
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class represents a log entry and contains some of the fields that are more important
+ * for querying logs action and are of more interest to maintainers.
+ */
+public class GeneralLogEntry {
+    private final String logName;
+    private final String severity;
+    private final String trace;
+    private final SourceLocation sourceLocation;
+    private final long timestamp;
+    @Nullable
+    private String textPayloadMessage;
+    @Nullable
+    private Map<String, Object> jsonPayloadMap;
+
+    public GeneralLogEntry(String logName, String severity, String trace, SourceLocation sourceLocation,
+                           long timestamp, String textPayloadMessage) {
+        this.logName = logName;
+        this.severity = severity;
+        this.trace = trace;
+        this.sourceLocation = sourceLocation;
+        this.timestamp = timestamp;
+        this.textPayloadMessage = textPayloadMessage;
+    }
+
+    public GeneralLogEntry(String logName, String severity, String trace, SourceLocation sourceLocation,
+                           long timestamp, Map<String, Object> jsonPayloadMap) {
```

Comment:
Rather than duplicating the constructor, why not just make a constructor with all parameters other than the payload, and pass in the payload via setter? The payloads are not `final`.

Reply:
Will change to passing payload via setter